### PR TITLE
AF-2638: Index page is not updated when changing models

### PR DESCRIPTION
**Pull request by jesuino**
[AF-2638](https://issues.redhat.com/browse/AF-2638)

SyncBeanManager does not override registered beans, hence we need to replace the page content instead register it again and again

### CODE REVIEW

**Comment by jesuino** [comment]
jenkins execute fdb

File: dashbuilder/dashbuilder-runtime/src/main/java/org/dashbuilder/client/perspective/generator/RuntimePerspectiveGenerator.java [hunk 1]
```diff
@@ -50,10 +52,17 @@
...
+            return createNewPerspective(layoutTemplate,
+                                 screen);
+        } else {
+            PerspectiveEditorScreenActivity screen = updateScreen(layoutTemplate);
+            return updatePerspective(layoutTemplate,
+                              screen);
+        }
     }
-
+
```
Comment (caponetto) [inline dashbuilder/dashbuilder-runtime/src/main/java/org/dashbuilder/client/perspective/generator/RuntimePerspectiveGenerator.java:65]:
```suggestion

```

File: dashbuilder/dashbuilder-runtime/src/main/java/org/dashbuilder/client/perspective/generator/RuntimePerspectiveGenerator.java [hunk 2]
```diff
@@ -95,5 +104,28 @@ private PerspectiveEditorActivity createNewPerspective(LayoutTemplate perspectiv
...
+    }
+    
+    private PerspectiveEditorActivity updatePerspective(LayoutTemplate layoutTemplate,
+                                                        PerspectiveEditorScreenActivity screen) {
+                             final SyncBeanDef<Activity> activity = activityBeansCache.getActivity(layoutTemplate.getName());
+                             final PerspectiveEditorActivity perspectiveEditorActivity = (PerspectiveEditorActivity) activity.getInstance();
+                             perspectiveEditorActivity.update(layoutTemplate,
+                                                              screen);
+                             return perspectiveEditorActivity;
+                         }
```
Comment (caponetto) [inline dashbuilder/dashbuilder-runtime/src/main/java/org/dashbuilder/client/perspective/generator/RuntimePerspectiveGenerator.java:123]: Please indent this piece of code.

File: dashbuilder/dashbuilder-runtime/src/main/java/org/dashbuilder/client/perspective/generator/RuntimePerspectiveGenerator.java [hunk 3]
```diff
@@ -95,5 +104,28 @@ private PerspectiveEditorActivity createNewPerspective(LayoutTemplate perspectiv
...
+    
+    
+    private boolean isANewPerspective(LayoutTemplate layoutTemplate) {
+        final IOCBeanDef<Activity> activity = activityBeansCache.getActivity(layoutTemplate.getName());
+        return activity == null;
```
Comment (caponetto) [inline dashbuilder/dashbuilder-runtime/src/main/java/org/dashbuilder/client/perspective/generator/RuntimePerspectiveGenerator.java:128]:
```suggestion
        return activityBeansCache.getActivity(layoutTemplate.getName()) == null;
```

**Review by caponetto — COMMENTED** [review]
Hi @jesuino, thanks for the PR!
I've left some comments, please check if they make sense.

**Comment by jesuino** [comment]
Thank you, @caponetto I addressed the changes in my latest commit

**Review by caponetto — APPROVED** [review]

**Comment by barboras7** [comment]
jenkins execute fdb

**Review by barboras7 — APPROVED** [review]
Works as expected, thanks @jesuino!

**Review by ederign — APPROVED** [review]

**Comment by jesuino** [comment]
jenkins execute fdb

**Comment by jesuino** [comment]
jenkins execute cdb

**Comment by jesuino** [comment]
The failed tests are unrelated, but I will try one more fdb to see if it gets green

org.kie.server.integrationtests.jbpm.NotificationSaveContentIntegrationTest.testSaveContentOnNotification[0: JAXB KieServicesConfiguration{transport=REST, serverUrl='http://localhost:33358/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.jbpm.NotificationSaveContentIntegrationTest.testSaveContentOnNotification[0: JAXB KieServicesConfiguration{transport=REST, serverUrl='http://localhost:33358/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.jbpm.NotificationSaveContentIntegrationTest.testSaveContentOnNotification[1: JSON KieServicesConfiguration{transport=REST, serverUrl='http://localhost:33358/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.jbpm.NotificationSaveContentIntegrationTest.testSaveContentOnNotification[2: XSTREAM KieServicesConfiguration{transport=REST, serverUrl='http://localhost:33358/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.jbpm.NotificationSaveContentIntegrationTest.testSaveContentOnNotification[2: XSTREAM KieServicesConfiguration{transport=REST, serverUrl='http://localhost:33358/kie-server-services/services/rest/server'}]
org.kie.server.integrationtests.jbpm.NotificationSaveContentIntegrationTest.testSaveContentOnNotification[3: JAXB KieServicesConfiguration{transport=JMS, serverUrl='null'}]
org.kie.server.integrationtests.jbpm.NotificationSaveContentIntegrationTest.testSaveContentOnNotification[3: JAXB KieServicesConfiguration{transport=JMS, serverUrl='null'}]
org.kie.server.integrationtests.jbpm.NotificationSaveContentIntegrationTest.testSaveContentOnNotification[4: JSON KieServicesConfiguration{transport=JMS, serverUrl='null'}]
org.kie.server.integrationtests.jbpm.NotificationSaveContentIntegrationTest.testSaveContentOnNotification[4: JSON KieServicesConfiguration{transport=JMS, serverUrl='null'}]
org.kie.server.integrationtests.jbpm.NotificationSaveContentIntegrationTest.testSaveContentOnNotification[5: XSTREAM KieServicesConfiguration{transport=JMS, serverUrl='null'}]
org.kie.server.integrationtests.jbpm.NotificationSaveContentIntegrationTest.testSaveContentOnNotification[5: XSTREAM KieServicesConfiguration{transport=JMS, serverUrl='null'}]

**Comment by jesuino** [comment]
jenkins execute fdb

**Comment by jesuino** [comment]
jenkins execute cdb

**Pull request merged**